### DISCUSSION TITLE
Unable to download nextflow on WSL (error:  line 322: java: command not found)

### DISCUSSION THREAD

**Discussion by bhoen1** [nextflow-io/nextflow#2737] [Q&A]
Hi all,

I am trying to download nextflow on wsl, but I am unable to. The error is related to java (see below).
I tried downloading nextflow with 3 methods:
- wget -qO- https://get.nextflow.io | bash
- curl -s https://get.nextflow.io | bash
- Installing nextflow by downloading on github.

All of them gave a Java related error. I re-downloaded java8.0 but this did not help.
<img width="603" alt="Schermafbeelding 2022-03-17 161647" src="https://user-images.githubusercontent.com/101816169/158834161-e30d861c-7c3e-4b68-aff1-3c996fd87c1b.png">

I also have Bash version >3.2
<img width="443" alt="Schermafbeelding 2022-03-17 173748" src="https://user-images.githubusercontent.com/101816169/158849639-10462fa9-eed7-4760-9a05-3a8e0d91b9f0.png">

Does someone have an idea how to solve this? Its my first time coding with linux, so I do not have any professional experience.

Thank you in advance!
Babette

**Comment by bentsherman** [comment]
Hi @bhoen1 , hopefully you figured out your problem by now but I'll answer just in case.

You should make sure you have java 11 or newer installed. You can install it with `sudo apt-get install default-jre`  or with sdkman:
```bash
curl -s "https://get.sdkman.io" | bash
source "${HOME}/.sdkman/bin/sdkman-init.sh"
sdk install java 11.0.12-open 
java -version
```

Installing Java on Windows doesn't count -- it needs to be installed inside your WSL environment.
